src/Npgsql/NpgsqlConnectionStringBuilder.cs

```diff
@@ -1737,5 +1766,34 @@ enum ReplicationMode
         /// </summary>
         Logical
     }
+
+    /// <summary>
+    /// Specifies server type preference.
+    /// </summary>
+    [Flags]
+    public enum TargetServerType : byte
```

Comment:
I found the [Flags] here a little bit confusing - we sometimes check `preferredType == TargetServerType.Primary`, sometimes `preferredType.HasFlag(TargetServerType.Primary)`... Is it maybe better to make this a regular enum?

Reply:
That's because whenever we're on the first cycle (looking for a preferred host), we can return it if the `preferredType` is either `TargetServerType.Primary` or `TargetServerType.PreferPrimary` (the same thing is for the secondary).

Reply:
Sure, I understand the logic - just found it a bit harder to follow.